Added recommendation for explicit typing of SD-JWTs (to address #267) (#275)
oauth-wg/oauth-selective-disclosure-jwt: draft-ietf-oauth-selective-disclosure-jwt.md

described in (#hash_function_claim). The SD-JWT payload MAY contain one or more selectively disclosable claims. It MAY also contain a Holder's public key or a reference
thereto, as well as further claims such as `iss`, `iat`, etc. as defined or
required by the application using SD-JWTs.

> Applications of SD-JWT SHOULD be explicitly typed using the `typ` header parameter, see (#explicit_typing) for more details.

Reviewer: ```suggestion

Applications of SD-JWT SHOULD be explicitly typed using the `typ` header parameter. See (#explicit_typing) for more details.
```

In some scenarios this behavior could be desirable,
but if it is not, Issuers need to support and Verifiers need to enforce Holder Binding.

## Explicit Typing {#explicit_typing}

> Section 3.11 of [@RFC8725] describes the use of explicit typing to prevent confusion attacks
> where one kind of JWT to be confused for another. SD-JWTs are also potentially

Reviewer: ```suggestion
Section 3.11 of [@RFC8725] describes the use of explicit typing to prevent confusion attacks
in which one kind of JWT is mistaken for another. SD-JWTs are also potentially
```
